cmd-tool/cloneset.lua

```diff
@@ -0,0 +1,78 @@
+local CloneSet = {}
+
+function CloneSet.captureCommandOutput(namespace, workloadName)
+    local command = "kubectl get cloneset.apps.kruise.io -n " .. namespace .. " -o yaml"
+
+    if workloadName ~= nil then
+        command = command .. " " .. workloadName
+    end
+
+    local handle = io.popen(command)
+    local output = handle:read("*a")
+    local success, exit_reason, exit_code = handle:close()
+
+    if not success then
+        return nil, exit_reason, exit_code
+    end
+
+    return output
+end
+
+function CloneSet.checkHealthWithTimeout(namespace, workloadName, timeout)
+    local lyaml = require("lyaml")
+
+    local function checkStatus()
+        local output = CloneSet.captureCommandOutput(namespace, workloadName)
+        local obj = lyaml.load(output)
+
+        local hs = { status = "Progressing", message = "Waiting for initialization" }
+
+        if obj then
+            -- Check if there is an "items" array, and if not, treat the object as a single Cloneset
+            local items = obj.items or { obj }
+
+            for _, item in ipairs(items) do
+                if item.status then
+                    if item.metadata.generation == item.status.observedGeneration then
+                        if item.spec.updateStrategy.paused == true or not item.status.updatedAvailableReplicas then
```

Comment:
i think this condition should be merged with L40  "Item.spec.updateStrategy.partition ~= 0", since they have exactly the same operation.

Reply:
i think we should keep both the conditions because both the conditions convey specific but different messages the first condition tells you that cloneset is paused and the other conditions tells that it needs manual intervention because something is broken!